src/main/java/org/osc/manager/rest/server/api/SecurityApis.java

```diff
@@ -0,0 +1,173 @@
+package org.osc.manager.rest.server.api;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.stream.Collectors;
+
+import javax.persistence.EntityManager;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.apache.log4j.Logger;
+import org.osc.manager.ism.api.IsmSecurityGroupApi;
+import org.osc.manager.ism.api.IsmSecurityGroupInterfaceApi;
+import org.osc.manager.ism.entities.SecurityGroupEntity;
+import org.osc.manager.ism.entities.SecurityGroupInterfaceEntity;
+import org.osc.manager.rest.server.SecurityManagerServerRestConstants;
+import org.osc.sdk.manager.element.ManagerSecurityGroupElement;
+import org.osc.sdk.manager.element.ManagerSecurityGroupInterfaceElement;
+import org.osc.sdk.manager.element.SecurityGroupInterfaceElement;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.transaction.control.TransactionControl;
+
+@Component(service = SecurityApis.class)
+@Path(SecurityManagerServerRestConstants.SERVER_API_PATH_PREFIX + "/securitygroups")
+@Consumes({ MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML })
+@Produces({ MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML })
+public class SecurityApis {
+    private static Logger LOG = Logger.getLogger(SecurityApis.class);
+    private EntityManager em;
+    private TransactionControl txControl;
+    private IsmSecurityGroupApi sgApi;
+    private IsmSecurityGroupInterfaceApi sgiApi;
+
+    public void init(EntityManager em, TransactionControl txControl) throws Exception {
+        this.em = em;
+        this.txControl = txControl;
+        this.sgApi = new IsmSecurityGroupApi(null, txControl, em);
+        this.sgiApi = new IsmSecurityGroupInterfaceApi(null, null, txControl, em);
+    }
+
+    @POST
+    public String createSecurityGroup(SecurityGroupEntity entity) throws Exception {
+        LOG.info(String.format("Creating security group  with name %s", "" + entity.getName()));
+        return this.sgApi.createSecurityGroup(entity.getName(), null, null);
+    }
+
+    @Path("/{sgId}")
+    @PUT
+    public SecurityGroupEntity updateSecurityGroup(@PathParam("sgId") Long sgId, SecurityGroupEntity entity)
+            throws Exception {
+        LOG.info(String.format("Updating the security group for id %s ", "" + Long.toString(sgId)));
+        this.sgApi.updateSecurityGroup(Long.toString(sgId), entity.getName(), null);
+        return entity;
+    }
+
+    @Path("/{sgId}")
+    @DELETE
+    public void deleteSecurityGroup(@PathParam("sgId") Long sgId) throws Exception {
+        LOG.info(String.format("Deleting the security group for id %s ", "" + Long.toString(sgId)));
+        this.txControl.required(new Callable<Void>() {
```

Comment:
Why do you need a transaction here but not on update or create?

Reply:
same as the other remark.

Reply:
This is returning error 500 due to rest layer returning before the transaction is completed.
I believe this is due to the plugin implementation, as plugin does the delete in two independent transaction steps, first for query in the independent transaction and second delete in another independent transaction. 
I have now fixed at the plugin level.